src/utils/utils.ts

```diff
@@ -1,9 +1,35 @@
 import { Dispatch, SetStateAction } from "react";
-import { querySearchService, urlWithEndSlash } from "../api/browse";
+import { querySearchService } from "../api/browse";
 import { SearchResponseModel } from "../models/dataset";
 import { FacetFilterModel } from "../models/facets";
 
-const CLIENT_URL = new URL(urlWithEndSlash(process.env.REACT_APP_CLIENT_URL!))
+/**
+ * Checks and cleans up URL if last character is or is not forward slash
+ * Duplicate of utils version because of infinite recursion error, needs refactoring
+ * @param url - String of URL
+ * @param endSlash - whether we want to have an end slash or not
+ * @returns URL as specified
+ */
+export const urlWithEndSlash = (url: string) => {
+  const lastCharSlash = url.endsWith("/");
+  return lastCharSlash ? url : url + "/"
+}
+
+export const CLIENT_URL = new URL(urlWithEndSlash(process.env.REACT_APP_CLIENT_URL!))
```

Comment:
This is a bit repetitive. You can add another helper function that takes the name of an env variable and returns a URL.

Reply:
thanks. actually the two functions could be converted to one-liners, i.e. directly return an expression instead of having a code block with return.